source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -106,12 +106,19 @@ namespace PlayFab
             }
 
             // pipeline failed to intake the event, create a response
-            std::shared_ptr<const PlayFabEmitEventRequest> playFabEmitRequest = std::dynamic_pointer_cast<const PlayFabEmitEventRequest>(request);
             auto playFabEmitEventResponse = std::make_shared<PlayFabEmitEventResponse>();
             playFabEmitEventResponse->emitEventResult = emitResult;
 
+            // TODO: since we are std::move-ing the request,
+            //  we actually lose this info to return to the user.
+            // Do users care about this object?
+            // the entire point of this callback would be for users to retry if something went wrong,
+            // but with this particular change they must re-construct the request
+            // ( BUT which request? we no longer know...)
+            std::shared_ptr<PlayFabEmitEventRequest> req;
```

Comment:
>req [](start = 53, length = 3)

you are passing a null shared_ptr in the callback below... is that intentional? why aren't you passing the request object provided in the parameter? Should the code be updated so that ownership is only transferred when to the eventbuffer on success? otherwise I think we still want to own that object to pass back out in the CallbackRequest